applications: nrf_desktop: Document MCUboot HW crypto for nRF54L SoC Series (#19356)
nrfconnect/sdk-nrf: applications/nrf_desktop/description.rst

.. nrf_desktop_fastpair_important_end

nRF54L MCUboot provisioning
===========================

nRF54L-based nRF Desktop devices enable hardware cryptography for the MCUboot bootloader.

> Public key used by the MCUboot for the application image validation is securely stored in hardware Key Management Unit (KMU).
> In this use case, application image is automatically signed by the |NCS| build system, but the public key is not automatically provisioned to device together with programming the bootloader and the application images (``west flash``).

Reviewer: ```suggestion
The public key that MCUboot uses to validate the application image is securely stored in the hardware Key Management Unit (KMU).
In this use case, the application image is automatically signed by the |NCS| build system.
However, the public key is not automatically provisioned to the device when programming the bootloader and the application images using the ``west flash`` command.
```

> Use ``west ncs-provision`` command to provision the MCUboot keys before programming the bootloader and the application images.

Reviewer: ```suggestion
To provision the MCUboot keys, use the ``west ncs-provision`` command before programming the bootloader and application images.
```

Make sure that the provisioned public key is generated from the private key that was used to sign the application image.
The private keys are stored in the application configuration directory of the board.

> Path to the used private key is defined by the ``SB_CONFIG_BOOT_SIGNATURE_KEY_FILE`` sysbuild Kconfig option.

Reviewer: ```suggestion
Path to the private key used is defined by the ``SB_CONFIG_BOOT_SIGNATURE_KEY_FILE`` sysbuild Kconfig option.
```
or maybe "used" isn't necessary at all if it's clear from the context?

Author: I can remove `used` here.

> Only one public key needs to be provisioned to an nRF Desktop device.

Reviewer: ```suggestion
You only need to provision one public key to an nRF Desktop device.
```

> Keep in mind that device must be provisioned again after it is erased.
> See :ref:`ug_nrf54l_developing_provision_kmu` documentation for nRF54L KMU provisionig details.

Reviewer: ```suggestion
Keep in mind that after erasing the device, you must provision it again.
For details, see :ref:`provisioning KMU for nRF54L devices<ug_nrf54l_developing_provision_kmu>`.
```

.. nrf_desktop_fastpair_important_end

nRF54L MCUboot provisioning
===========================

nRF54L-based nRF Desktop devices enable hardware cryptography for the MCUboot bootloader.
The public key that MCUboot uses to validate the application image is securely stored in the hardware Key Management Unit (KMU).
In this use case, the application image is automatically signed by the |NCS| build system.
However, the public key is not automatically provisioned to the device when programming the bootloader and the application images using the ``west flash`` command.

To provision the MCUboot keys, use the ``west ncs-provision`` command before programming the bootloader and application images.
Make sure that the provisioned public key is generated from the private key that was used to sign the application image.
The private keys are stored in the application configuration directory of the board.
Path to the private key is defined by the ``SB_CONFIG_BOOT_SIGNATURE_KEY_FILE`` sysbuild Kconfig option.
You only need to provision one public key to an nRF Desktop device.

> Keep in mind that after erasing the device, you must provision it again.

Reviewer 2: ```suggestion
.. note::
    If you are erasing the device, you must provision it again.
```
I think this could be a note since this is not part of the provisioning procedure above?, and if user erases the device, he need to provision the device again. If yes, could be added after line no:970.

Author: Making it as a note sounds like a good idea. Changed.

Author: Rephrased the note a bit to improve readability too

.. nrf_desktop_fastpair_important_end

> nRF54L MCUboot provisioning

Reviewer 3: Can we avoid target names in section titles? You can find a few suggestions below:

```
MCUboot provisioning using the key storage peripheral
```
or
```
MCUboot provisioning using the KMU peripheral
```
or
```
MCUboot provisioning using the HW storage
```

The KMU is also available for nRF53 Series.

And then we can list supported targets or even provide a reference to the list that is maintained in the general MCUboot documentation

Author: I intentionally prefer to keep `nRF54L` in the section name as currently it's the only SoC Series that supports the feature. Even Kconfigs and build system have SoC Series dependencies. I am worried that making the documentation more generic could lead to false impression that other SoC Series might also support these features.

Author: Also, using `nRF54L` in the title allows to skip the section for other SoC Series users without reading.

===========================

nRF54L-based nRF Desktop devices enable hardware cryptography for the MCUboot bootloader.
The public key that MCUboot uses to validate the application image is securely stored in the hardware Key Management Unit (KMU).
In this use case, the application image is automatically signed by the |NCS| build system.
However, the public key is not automatically provisioned to the device when programming the bootloader and the application images using the ``west flash`` command.

> To provision the MCUboot keys, use the ``west ncs-provision`` command before programming the bootloader and application images.

Reviewer 3: Wouldn't it make more sense to add this documentation section to general MCUboot documentation and only provide a link here?

Author: It would be a bit more complicated to make a generic documentation for MCUboot (using KMU is controlled with a Kconfig option, multiple public keys might be used etc.). I would prefer to keep this simple docs in scope of nRF Desktop for NCS 2.9. After MCUboot bootloader docs are improved, we could remove part of this documentation and replace it with a link.

Make sure that the provisioned public key is generated from the private key that was used to sign the application image.
The private keys are stored in the application configuration directory of the board.
Path to the private key is defined by the ``SB_CONFIG_BOOT_SIGNATURE_KEY_FILE`` sysbuild Kconfig option.
You only need to provision one public key to an nRF Desktop device.

> For details, see :ref:`provisioning KMU for nRF54L devices <ug_nrf54l_developing_provision_kmu>`.

Reviewer 4: Didn't we agree to  place here commends on how to provision dk?

Author: We discussed this idea, but eventually we desisted from doing it. More provisioning methods might be introduced in the future (I guess some of them would be done through GUI). Alternatively the provisioning might be done automatically together with programming the board. I would prefer not to maintain this doc part in scope of nRF Desktop.

The command is described on the linked `provisioning KMU for nRF54L devices` page. I could also add a single key command variant there to make it more straightforward for nRF Desktop users.
